### PR TITLE
Code validated by luacheck

### PR DESCRIPTION
**Pull request by pvinchon**
[luacheck](https://github.com/mpeterv/luacheck) is a simple tool that checks the validity of the code. Think of it as [JSLint](http://www.jslint.com/) for lua.

This allowed me to fix unused arguments / variables, undefined variables, and other simple errors.

### CODE REVIEW

File: src/torch/1-split-encoder-classifier/split-encoder-classifier.lua [hunk 1]
```diff
@@ -45,12 +47,12 @@ print(encoder)
...
 -- create classifier
 local classifier = net:clone()
-for i = 1, 21 do
+for _ = 1, 21 do
   classifier:remove(1)
 end
 
 local outputEnc = encoder:forward(im)
-local lossClassifier, outputClassifier = classifier:forward(outputEnc[1]):view(-1):float():sort(true)
+local _, outputClassifier = classifier:forward(outputEnc[1]):view(-1):float():sort(true)
```
Comment (paucarre) [inline src/torch/1-split-encoder-classifier/split-encoder-classifier.lua]: I wouldn't hide that. it's clear to have a name, even though we don't use it

File: src/torch/3-train-regression-bounding-box/train-regression-bounding-box.lua [hunk 2]
```diff
@@ -19,7 +21,32 @@ local inputsBatch = torch.Tensor(batchSize, 384, 11, 11):cuda()
...
+    end
+    gradParameters:zero()
+    local outputs = model:forward(inputsBatchesLocal)
+    local f = criterion:forward(outputs, outputsBatchLocal)
+    local df_do = criterion:backward(outputs, outputsBatchLocal)
+    model:backward(inputsBatchesLocal, df_do)
+    return f, gradParameters
+  end
+
+  local _, fx = optim.sgd(feval, parameters, optimState)
```
Comment (paucarre) [inline src/torch/3-train-regression-bounding-box/train-regression-bounding-box.lua]: I wouldn't hide variables, even if we don't use them

File: src/torch/5-train-classification/train-classification.lua [hunk 3]
```diff
@@ -92,62 +85,70 @@ function getTestError(model)
...
-    model:backward(inputsBatch, df_do)
+    local outputs = model:forward(inputsBatchLocal)
+    local f = criterion:forward(outputs, targetsBatchLocal)
+    local df_do = criterion:backward(outputs, targetsBatchLocal)
+    model:backward(inputsBatchLocal, df_do)
     collectgarbage()
     return f, gradParameters
   end
-  local x, fx = optim.sgd(feval, parameters, optimState)
+  local _, fx = optim.sgd(feval, parameters, optimState)
```
Comment (paucarre) [inline src/torch/5-train-classification/train-classification.lua]: same, I wouldn't hide

File: src/torch/6-bboxlib/bboxlib.lua [hunk 4]
```diff
@@ -53,29 +56,15 @@ function toOutputCoordinates(x, y, reduction, xdelta, ydelta)
...
-
-function toImageCoordinate(coord)
+local function toImageCoordinate(coord)
   return ((((((coord - 1 + 2 + 10) * 2) + 2) * 2) + 2) * 4) + 10 + 7 - 224
 end
 
---  Equations:
---    Ow = (Iw - delta) / reduction --> delta = Iw - (Ow * reduction)
---    Ox = (Ix - delta) / reduction
---    Ix = (Ox * reduction) + delta
```
Comment (paucarre) [inline src/torch/6-bboxlib/bboxlib.lua]: why was that removed?
Even if it's not used, in this case it shouldn't be removed as it might be used later on.

File: src/torch/6-bboxlib/bboxlib.lua [hunk 5]
```diff
@@ -162,28 +140,30 @@ function cleanBoundingBox(x, y, imageW, imageH)
...
+end
 
-function getEncodedInput(input)
+local function getEncodedInput(input)
   local encoder = loadEncoder()
   local encodedInput = encoder:forward(input)[2]
   return encodedInput, input:size()[3], input:size()[2]
 end
 
-function getEncodedLowInput(input)
```
Comment (paucarre) [inline src/torch/6-bboxlib/bboxlib.lua]: Not used, but I wouldn't remove it. Can be used later on.

File: src/torch/6-bboxlib/bboxlib.lua [hunk 6]
```diff
@@ -194,37 +174,7 @@ function bboxlib.loadImageFromFile(imagePath)
...
-  for file in lfs.dir(folder) do
-    if (lfs.attributes(folder .. '/' .. file, "mode") == "file") then
-      table.insert(files, folder .. '/' .. file)
-    end
-  end
-  return files
-end
-
-function encodeAndNormalizeImage(croppedScaledImage)
-  local croppedScaledImage = tiefvision_commons.loadImage(croppedScaledImage)
```
Comment (paucarre) [inline src/torch/6-bboxlib/bboxlib.lua]: Why is this removed?

File: src/torch/6-bboxlib/bboxlib.lua [hunk 7]
```diff
@@ -262,31 +212,26 @@ function getBboxes(input, reduction)
...
 
-
-function getCroppedImage(input, reduction)
+local function getCroppedImage(input)
   local scale = getInitialScales(input)[2]
   local scaledImage = getScaledImages(input, { scale })[1]
-  local width, height = scaledImage:size()[3], scaledImage:size()[2]
-  local bboxes, probabilities, probDensity = getBboxes(scaledImage, reduction)
-  local xminNew, yminNew, xmaxNew, ymaxNew = getExpectedBoundingBox(bboxes, probabilities, width, height, reduction, xdelta, ydelta)
+  local bboxes, _ = getBboxes(scaledImage)
```
Comment (paucarre) [inline src/torch/6-bboxlib/bboxlib.lua]: I wouldn't hide that

File: src/torch/6-bboxlib/bboxlib.lua [hunk 8]
```diff
@@ -262,31 +212,26 @@ function getBboxes(input, reduction)
...
       if (probabilities[y][x] > 0.85) then
-        local xmin, ymin, xmax, ymax = getImageBoundingBox(x, y, bboxMinx, bboxMiny, bboxMaxx, bboxMaxy, width, height, reduction)
+        local xmin, ymin, xmax, ymax = getImageBoundingBox(x, y, bboxMinx, bboxMiny, bboxMaxx, bboxMaxy, width, height)
         bboxes[i] = { xmin, ymin, xmax, ymax, probabilities[y][x], x, y }
         i = i + 1
       end
     end
   end
-  return bboxes, probabilities, probDensity
+  return bboxes, probabilities
```
Comment (paucarre) [inline src/torch/6-bboxlib/bboxlib.lua]: probDensity is potentially useful later on.

**Review by paucarre — COMMENTED** [review]
Very good refactoring, but a few comments.

**Comment by Plippe** [comment]
Unused code should always be deleted. Current code is smaller, making it easier to understand, and the deleted code isn't lost as it can still be found in the git history.

**Review by paucarre — APPROVED** [review]

File: src/torch/12-deeprank-train/deeprank-train.lua [hunk 9]
```diff
@@ -26,7 +27,7 @@ function getSimilarityModel()
   --leftImage:add(nn.SpatialConvolutionMM(512, 126, 1, 1, 1, 1, 0, 0):cuda())
```
Comment (emersonloureiro) [inline src/torch/12-deeprank-train/deeprank-train.lua]: Should we also not kill lines which are commented out, and which are not actual comments?

**Review by emersonloureiro — APPROVED** [review]
Really nice cleanup.

File: src/torch/1-split-encoder-classifier/split-encoder-classifier.lua [hunk 10]
```diff
@@ -69,17 +71,17 @@ end
 
 -- Test encoder reduction
 local fakeIm = torch.Tensor(3, 224, 224):cuda()
-local outputEnc = encoder:forward(fakeIm)
+outputEnc = encoder:forward(fakeIm)
```
Comment (Plippe) [inline src/torch/1-split-encoder-classifier/split-encoder-classifier.lua]: `outputEnc` is declared above, redeclaring variables is flagged as an error by luacheck

File: src/torch/12-deeprank-train/deeprank-train.lua (hunk 9 above)
Comment (Plippe) [inline src/torch/12-deeprank-train/deeprank-train.lua]: Not sure, my eyes skip them when I am not interested in this part of the code, but it does add value when you are looking for things that have been tried in the past.

Your call

Comment (emersonloureiro) [inline src/torch/12-deeprank-train/deeprank-train.lua]: Not fussy about it, honestly, so will leave it up to you. In general, I think people would not look at that sort of thing for reference, so I usually clean it up. If you think it may be beneficial for this project - I've never worked on it, so hard for me to tell - then it's more than fine to leave it there.
